Bump Angular to 11.0

oliverguenther opened the pull request:
- [x] Fix issues due to the migration to TypeScript 4.0
- [x] Replace node-sass with sass and fix compilation errors
- [x] Moved some overriden property getters into actual properties due to TS2611 strictness compilation errors

oliverguenther committed:
Bump Angular to 11

oliverguenther committed:
Bump fullcalendar

oliverguenther committed:
Bump uirouter

oliverguenther committed:
bump typescript and rxjs

fixup! bump typescript and rxjs

oliverguenther committed:
Run angular update migrations

oliverguenther committed:
Fix deprecated options in serve, they are kept in builder

oliverguenther committed:
Replace node-sass with sass and fix errors

oliverguenther committed:
Include all ts files again to fix warnings

We have scripts that should be contained but are only loaded dynamically. It seems this trips up TS

to print warnings that these files are unused, which they really aren't.

oliverguenther committed:
Address typescript compilation errors and warnings due to 4.0 upgrade

openprojectci commented:
<!--
  0 Errors
  6 Warnings: <a href="https://github.com/op...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          6 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/0f263aa1a2c5ffe183d1663f0d356733547841fa/frontend/src/app/components/work-packages/work-package-comment/wp-comment-field.component.ts#L36">frontend/src/app/components/work-packages/work-package-comment/wp-comment-field.component.ts#L36</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/0f263aa1a2c5ffe183d1663f0d356733547841fa/frontend/src/app/modules/common/autocomplete/create-autocompleter.component.ts#L42">frontend/src/app/modules/common/autocomplete/create-autocompleter.component.ts#L42</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/0f263aa1a2c5ffe183d1663f0d356733547841fa/frontend/src/app/modules/common/autocomplete/version-autocompleter.component.ts#L37">frontend/src/app/modules/common/autocomplete/version-autocompleter.component.ts#L37</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/0f263aa1a2c5ffe183d1663f0d356733547841fa/frontend/src/app/modules/common/editable-toolbar-title/editable-toolbar-title.component.ts#L46">frontend/src/app/modules/common/editable-toolbar-title/editable-toolbar-title.component.ts#L46</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/0f263aa1a2c5ffe183d1663f0d356733547841fa/frontend/src/app/modules/fields/edit/field-types/multi-select-edit-field.component.ts#L37">frontend/src/app/modules/fields/edit/field-types/multi-select-edit-field.component.ts#L37</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/0f263aa1a2c5ffe183d1663f0d356733547841fa/frontend/src/app/modules/fields/edit/field-types/project-status-edit-field.component.ts#L35">frontend/src/app/modules/fields/edit/field-types/project-status-edit-field.component.ts#L35</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

oliverguenther committed:
Link to styles.css not js with updated sass

oliverguenther committed:
Only reposition dropdown if appendTo is set

oliverguenther committed:
Initialize resource property in edit field component

Correctly update field component and handler properties

Always use the changeset schema

Fix removed property in custom text

oliverguenther commented:
There's an unrelated Ruby unit test flickering

ulferts reviewed (approved):
:+1:
